types.go

```diff
@@ -42,18 +44,25 @@ func (f *OneBlockFile) Data(ctx context.Context, s dstore.Store) ([]byte, error)
 }
 
 func (f *OneBlockFile) downloadFile(ctx context.Context, s dstore.Store) error {
-	out, err := s.OpenObject(ctx, f.name)
-	if err != nil {
-		return err
-	}
-	defer out.Close()
+	var err error
```

Comment:
It's not obvious that we will keep the last error and return it at the end if loop through all of them and nothing worked.

I would add a comment that explain that.